### ISSUE TITLE
tiled::FiniteTileLayer::get_tile returns empty tiles

### ISSUE THREAD

**Issue by jruiss** [mattyhall/rs-tiled#236]
https://docs.rs/tiled/latest/tiled/struct.FiniteTileLayer.html#method.get_tile

Resulting in a single layer with just a couple of tiles overrides all layers under with tile id 0 when rendered.

**Comment by aleokdev** [comment]
Ah darn, I forgot to reply to this.
I don't quite understand the issue. Could you rephrase it, or send an instance of the error we could reproduce?

**Comment by IoIxD** [comment]
The data of each tile returned by the function is empty.

state.rs
```
. . .
    pub fn for_each_tile(&self, f: &mut dyn FnMut(LayerTile<'_>, i32, i32)) {
. . .
                tiled::LayerType::TileLayer(layer) => {
                    match layer {
                        TileLayer::Finite(a) => {                            
                            for y in 0..a.height() {
                                for x in 0..a.width() {
                                    if let Some(a) = a.get_tile(x as i32, y as i32) {
                                        f(a, x as i32, y as i32)
                                    }
                                }
                            }
                        },
                        TileLayer::Infinite(_) => todo!(),
                    }
                },
. . .
```

render.rs
```
. . .
                st.for_each_tile(&mut |f, x, y| {
                    let tile = f.get_tile().unwrap();
                    println!("{:#?}",tile.deref());
                    match &tile.image {
                        Some(b) => {
                            println!("{:#?}",b);
                            let t = a.get(&b.source.to_str().unwrap().to_string()).unwrap();
                            d2.draw_texture(t, x, y, Color::WHITE)
                        },
                        None => {}
                    };
                });
. . .
```

output:
```
TileData {
    image: None,
    properties: {},
    collision: None,
    animation: None,
    tile_type: None,
    probability: 0.0,
}
TileData {
    image: None,
    properties: {},
    collision: None,
    animation: None,
    tile_type: None,
    probability: 0.0,
}
TileData {
    image: None,
    properties: {},
    collision: None,
    animation: None,
    tile_type: None,
    probability: 0.0,
}
TileData {
    image: None,
    properties: {},
    collision: None,
    animation: None,
    tile_type: None,
    probability: 0.0,
}
```

**Comment by aleokdev** [comment]
What's your tileset like? It is possible that the data is empty because the tiles in the tileset don't have any properties/image/collision/etc set, which is the default behavior for any new tileset. Unless you are working with image collection tilesets you shouldn't need to access the TileData at all, just the tile ID and the tileset it is from.

**Comment by IoIxD** [comment]
Sorry, a lot of stuff happened, but I'm getting back to my project now.
Here's my tileset:

[test.zip](https://github.com/mapeditor/rs-tiled/files/10843940/test.zip)

**Comment by IoIxD** [comment]
Indeed there is no tile information. 
I expected to be able to get image data from the tiles, hence my confusion, but apparently the file format doesn't actually store that information and i have to store that information myself. I misunderstood the crate.

**Comment by aleokdev** [comment]
> Indeed there is no tile information. 
> I expected to be able to get image data from the tiles, hence my confusion, but apparently the file format doesn't actually store that information and i have to store that information myself. I misunderstood the crate.

I see. I understand that you thought that the crate split the image for each tile. If so you are correct in that the fileformat does not contain per-tile image data. What the user is expected to do is to use the tileset image instead, and by using UV coordinates, texture splitting or other mechanisms determine the tile image.

The reasoning behind this is that the crate _does not even load the image_ for simplicity. Normally you give the path of the image to the engine or framework you are using to load the image from there, which is more extensible as there is no middle representation of the image in between both libraries.

**Comment by aleokdev** [comment]
In any case I recommend you take a look at the SFML example provided to see how images are normally treated and loaded, regardless of the framework you are using 🙂
